chat-widget/src/components/livechatwidget/common/startChat.ts

```diff
@@ -88,34 +112,39 @@ const initStartChat = async (chatSDK: any, dispatch: Dispatch<ILiveChatWidgetAct
                 }
             });
             isStartChatSuccessful = false;
+            // Resetting the widget state to Closed, for recent introduction of OC rate limiting(429 Error) 
+            // TODO : How to diplay a proper UI message to customer to try after sometime at this point - cool down scenario
+            dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Closed });
+            return;
         }
+        
+        // New adapter creation
         const newAdapter = await createAdapter(chatSDK);
         setAdapter(newAdapter);
 
         const chatToken = await chatSDK.getChatToken();
+        dispatch({ type: LiveChatWidgetActionType.SET_CHAT_TOKEN, payload: chatToken });
         newAdapter?.activity$?.subscribe(createOnNewAdapterActivityHandler(chatToken?.chatId, chatToken?.visitorId));
-        if (!persistedState) {
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            if ((chatSDK as any)?.getVoiceVideoCalling) {
-                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                const chatToken: any = await chatSDK?.getChatToken();
-                dispatch({ type: LiveChatWidgetActionType.SET_CHAT_TOKEN, payload: chatToken });
-            }
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
-            const liveChatContext: any = await chatSDK?.getCurrentLiveChatContext();
-            dispatch({ type: LiveChatWidgetActionType.SET_LIVE_CHAT_CONTEXT, payload: liveChatContext });
 
-            await setPostChatContextAndLoadSurvey(chatSDK, dispatch);
-
-            await updateSessionDataForTelemetry(chatSDK, dispatch);
-
-            // Set app state to Active
-            if (isStartChatSuccessful) {
-                dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Active });
-            }
-        } else {
+        if (persistedState) {
             dispatch({ type: LiveChatWidgetActionType.SET_WIDGET_STATE, payload: persistedState });
             await setPostChatContextAndLoadSurvey(chatSDK, dispatch, true);
+            return;
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const liveChatContext: any = await chatSDK?.getCurrentLiveChatContext();
+        dispatch({ type: LiveChatWidgetActionType.SET_LIVE_CHAT_CONTEXT, payload: liveChatContext });
+
+        // Set post chat context in state, no survey load
+        await setPostChatContextAndLoadSurvey(chatSDK, dispatch);
+
+        // Updating chat session detail for telemetry
+        await updateSessionDataForTelemetry(chatSDK, dispatch);
```

Comment:
Do we need this in case of persisted chat?

Reply:
no, `persistedState` already has them, so not needed to do this steps again.